openshift-kube-apiserver/openshiftkubeapiserver/patch_handlerchain.go

```diff
@@ -95,3 +109,59 @@ func translateLegacyScopeImpersonation(handler http.Handler) http.Handler {
 		handler.ServeHTTP(w, req)
 	})
 }
+
+// withWatchRateLimit rate limits the establishment of watches
+func withWatchRateLimit(handler http.Handler) http.Handler {
+	// This allows over 1000 watch establishments a minute.  That is a fairly large number for steady state,
+	// but a manageable number that is lower than the the 30,000 per minute observed during disruptive events
+	// on a CI cluster.
+	// This effectively creates an upper bound on secret mounts of 10,000 because of a ten minute watch timeout.
+	// TODO I need to think about how many are practical before merging.
+	//  I could also image having this control in place for the first ten minutes of a kube-apiserver's life and
+	//  then relaxing it considerably or removing it altogether after that time.
```

Comment:
How about a scenario where a server `A` restarts, the active watches resident on `A` get killed and they relocate to server `B` in a land rush.